### PR TITLE
 Visitor can see different products on a restaurants menu 

### PR DESCRIPTION
**Pull request by robin-lillqvist**
# [PT Story](https://www.pivotaltracker.com/story/show/171561436)

## Changes proposed in this pull request:
* Present menu with menu-items fetched from JSON-file, in wait of a real backend api to fetch from.
## What I have learned working on this feature:
* Fetching data from Json file and populating html elements with data. Also to not overreach in the future and instead ask coach in case we are not sure what to do.
## Packages/Gems added
-
## Screenshots (Only if client side story)
Screenshot 2020-03-04 at 15 17 55

```
As a visitor,
In order to get an overview of the restaurant's products
I would like to be able to see a menu
```
## Included in this pull request: 
- trying to fix menu listing
- trying to show menu items on site
- adds early tests for menu
- started to work on app.jsx
- added first tests

### CODE REVIEW

File: public/menu.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[
+  {
```
Comment (oliverochman) [inline public/menu.json]: this should be wrapped in a products key

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  componentDidMount () {
+    axios.get('menu.json').then(response => {
+      this.setState({
+        foodItems: response.data
+      })
+    })
+  }
+
+  render () {
+    const foodItems = this.state.foodItems
```
Comment (oliverochman) [inline src/App.jsx]: please extract this to a separate component

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 2 above)
Comment (robin-lillqvist) [reply src/App.jsx]: Fixed this.

File: public/menu.json (hunk 1 above)
Comment (robin-lillqvist) [reply public/menu.json]: Fixed this.

File: cypress/integration/userCanSeeMeals.feature.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+describe('User can see Meals', () => {
+    before(function () {
+        cy.visit('/')
```
Comment (oliverochman) [inline cypress/integration/userCanSeeMeals.feature.js]: you are not stubbing out the requests that fetches the product data, that is why it is failing on semaphore

**Review by oliverochman — COMMENTED** [review]
please add the `cy:run` script to the package.json

File: cypress/integration/userCanSeeMeals.feature.js (hunk 3 above)
Comment (robin-lillqvist) [reply cypress/integration/userCanSeeMeals.feature.js]: Fixed this now we think!

**Comment by robin-lillqvist** [comment]
@oliverochman could you please take another look? 👍

File: src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from 'react'
+// import axios from 'axios'
```
Comment (faraznaeem) [inline src/App.jsx]: delete the commented out code

File: src/App.jsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import ShowMenu from './components/ShowMenu'
+import './App.css'
+
+class App extends Component {
+
+
```
Comment (faraznaeem) [inline src/App.jsx]: remove extra lines

File: src/components/ShowMenu.jsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react'
+import axios from 'axios'
+
+export default class ShowMenu extends Component {
```
Comment (faraznaeem) [inline src/components/ShowMenu.jsx]: move the export default to the bottom of the class, you need to stay consistent

File: public/index.html [hunk 7]
```diff
@@ -10,34 +10,15 @@
...
-      It will be replaced with the URL of the `public` folder during the build.
-      Only files inside the `public` folder can be referenced from the HTML.
-
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.css"/>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.js"></script>
```
Comment (faraznaeem) [inline public/index.html]: I would advise you to use react-semantic ui, and not the cdn as in the portfolio challenge

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: public/index.html (hunk 7 above)
Comment (robin-lillqvist) [reply public/index.html]: I tried to add that, but they still wanted me to have one row in the html.

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**